lib/services/saml.go

```diff
@@ -103,6 +103,10 @@ func ValidateSAMLConnector(sc types.SAMLConnector, rg RoleGetter) error {
 	if rg != nil {
 		for _, mapping := range sc.GetAttributesToRoles() {
 			for _, role := range mapping.Roles {
+				if strings.ContainsRune(role, '$') {
```

Comment:
In the linked issue you said that we could replace `$$` with `$` and assume this was a valid role name (and not templating).
It was not clear that we should do that.

Just asking to ensure we didn't forget that.

Reply:
I chose not to do that in the end - it started assuming too much about the implementation and I wondered if there should be other substitutions I was not aware of. However I think it becomes moot when I implement r0mant's feedback to use `utils.ContainsExpansion` - the decision is now delegated.

Reply:
Actually I'm wrong on that - if I use `utils.ContainsExpansion`, it will return false if the name contains `$$`, so we will try to check if the role name with `$$` is present and will fail. I will have to do the replacement in this case and check the role name.